drivers/include/ina3221.h

```diff
@@ -0,0 +1,823 @@
+ * @return      -EINVAL, if NULL pointer was passed
+ * @return      -INA3221_I2C_ERROR, if I2C bus acquirement failed
+ * @return      -INA3221_BAD_MANUF_ID, if manufacturer ID does not match
+ * @return      -INA3221_DIE_ID, if DIE ID does not match
+ * @return      -INA3221_RESET_FAILED, if reset failed
+ * @return      -INA3221_CONFIG_FAILED, if configuration could not be applied
+ * @return      @see i2c_read_regs
+ * @return      @see i2c_write_regs
+ */
+int ina3221_init(ina3221_t *dev, const ina3221_params_t *params);
+
+#if defined(MODULE_PERIPH_GPIO_IRQ) || defined(DOXYGEN)
+/**
+ * @brief Enable alert callback and argument for alert @p alert
+ *
+ * @param[in,out] dev       Device handle
+ * @param[in] alert         Alert index
+ * @param[in] cb            Alert callback
+ * @param[in] arg           Alert callback argument
+ *
+ * @return INA3221_OK, on success
+ * @return -ERANGE, if @p alert was out of bounds
+ * @return -ENUTSUP, if alert pin was initialized with GPIO_UNDEF
+ */
+int ina3221_enable_alert(ina3221_t *dev, ina3221_alert_t alert, ina3221_alert_cb_t cb, void* arg);
+#endif
+
+#if defined(MODULE_PERIPH_GPIO_IRQ) || defined(DOXYGEN)
+/**
+ * @brief Disable alert callback and argument for alert @p alert
+ *
+ * @param[in,out] dev       Device handle
+ * @param[in] alert         Alert index
+ *
+ * @return INA3221_OK, on success
+ * @return -ERANGE, if @p alert was out of bounds
+ * @return -ENUTSUP, if alert pin was initialized with GPIO_UNDEF
+ */
+int ina3221_disable_alert(ina3221_t *dev, ina3221_alert_t alert);
+#endif
+
+/**
+ * @brief Write configuration register value to configuration register
+ *
+ * @param[in,out]   dev Device handle
+ * @param[in]       cfg Configuration register value in host byte order
+ *
+ * @return      INA3221_OK, on success
+ * @return      -INA3221_I2C_ERROR, if I2C bus acquirement failed
+ * @return      @see i2c_write_regs
+ */
+int ina3221_set_config(ina3221_t* dev, uint16_t cfg);
+
+/**
+ * @brief Read currently saved configuration register value
+ * 
+ * @param[in]       dev Device handle
+ * @param[out]      cfg Pointer to configuration register output value in host byte order
+ * 
+ * @return      INA3221_OK
+ */
+int ina3221_get_config(const ina3221_t* dev, uint16_t* cfg);
+
+/**
+ * @brief Enable channels
+ * 
+ * @param[in,out]   dev Device handle
+ * @param[in]       ech Channel enable flags
+ * 
+ * @return      INA3221_OK, on success
+ * @return      -INA3221_I2C_ERROR, if I2C bus acquirement failed
+ * @return      @see i2c_read_regs
+ * @return      @see i2c_write_regs
+ */
+int ina3221_set_enable_channel(ina3221_t *dev, ina3221_enable_ch_t ech);
+
+/**
+ * @brief Read which channels are currently enabled
+ * 
+ * @param[in]       dev Device handle
+ * @param[out]      ech Pointer to enabled channels output variable
+ * 
+ * @return      Number of enabled channels
+ */
+int ina3221_get_enable_channel(const ina3221_t *dev, ina3221_enable_ch_t* ech);
```

Comment:
I think again this would be easier to use by providing a `static inline` wrapper function that does the decoding of the bitmask. At compile time, the compiler will be able to garbage collect dead code. E.g. if I'm only interested in whether or not channel 2 is active, the compiler will see that the writes to variables where is stored whether channel1 and channel 3 is enabled are never read afterwards, and it will optimized that code out.